migrator/models/domain.py

```diff
@@ -0,0 +1,177 @@
+from sqlalchemy import orm
+from sqlalchemy_utils.types.encrypted.encrypted_type import (
+    AesGcmEngine,
+    StringEncryptedType,
+)
+
+from migrator.extensions import config
+from migrator.models.common import (
+    Action,
+    RouteType,
+    RouteModel,
+    CertificateModel,
+    ChallengeModel,
+    AcmeUserV2Model,
+    OperationModel,
+    OperationState,
+)
+
+convention = {
+    "ix": "idx_%(column_0_label)s",
+    "uq": "uq_%(table_name)s_%(column_0_name)s",
+    "ck": "ck_%(table_name)s_%(constraint_name)s",
+    "fk": "fk_%(table_name)s_%(column_0_name)s_%(referred_table_name)s",
+    "pk": "pk_%(table_name)s",
+}
+
+metadata = sa.MetaData(naming_convention=convention)
+DomainModel = declarative.declarative_base(metadata=metadata)
+
+
+def db_encryption_key():
+    return config.DOMAIN_DATABASE_ENCRYPTION_KEY
+
+
+class DomainAlbProxy(DomainModel):
+    __tablename__ = "alb_proxies"
+
+    alb_arn = sa.Column(sa.Text, primary_key=True)
+    alb_dns_name = sa.Column(sa.Text)
+    listener_arn = sa.Column(sa.Text)
+
+
+class DomainRoute(DomainModel, RouteModel):
+    __tablename__ = "routes"
+
+    instance_id = sa.Column("guid", sa.Text, primary_key=True)
+    state = sa.Column(sa.Text, nullable=False, index=True)
+    domains = sa.Column(postgresql.ARRAY(sa.Text))
+    challenge_json = sa.Column(postgresql.BYTEA)
+    user_data_id = sa.Column(sa.Integer)
+    alb_proxy_arn = sa.Column(sa.Text)
+    alb_proxy: DomainAlbProxy = orm.relationship(
+        DomainAlbProxy,
+        foreign_keys=[alb_proxy_arn],
+        primaryjoin="DomainRoute.alb_proxy_arn == DomainAlbProxy.alb_arn",
+    )
+    certificates: List["DomainCertificate"] = orm.relationship(
+        "DomainCertificate",
+        order_by="desc(DomainCertificate.expires)",
+        primaryjoin="(foreign(DomainCertificate.route_guid)) == DomainRoute.instance_id",
+        backref="route",
+    )
+    operations: List["DomainOperation"] = orm.relationship(
+        "DomainOperation", backref="route", lazy="dynamic"
+    )
+    acme_user_id = sa.Column(sa.Integer, sa.ForeignKey("acme_user_v2.id"))
+    route_type = RouteType.ALB
+
+    def domain_external_list(self):
+        """to match CdnRoute"""
+        return self.domains
+
+
+class DomainCertificate(DomainModel, CertificateModel):
+    __tablename__ = "certificates"
+
+    id = sa.Column(sa.Integer, sa.Sequence("certificates_id_seq"), primary_key=True)
+    created_at = sa.Column(postgresql.TIMESTAMP)
+    updated_at = sa.Column(postgresql.TIMESTAMP)
+    deleted_at = sa.Column(postgresql.TIMESTAMP, index=True)
+    route_guid = sa.Column(sa.Text)
+    domain = sa.Column(sa.Text)
+    # cert_url is the Let's Encrypt URL for the certificate
+    cert_url = sa.Column(sa.Text)
+    # certificate is the actual body of the certificate chain
+    # this was used by the old broker, but the renewer uses fullchain_pem and leaf_pem instead
+    certificate = sa.Column(postgresql.BYTEA)
```

Comment:
Same note as above - kept for backward compatibility reasons?

Reply:
same answer as above